frontend/src/metabase/components/CheckBox.styled.js

```diff
@@ -0,0 +1,75 @@
+import React from "react";
+import styled, { css } from "styled-components";
+import _ from "underscore";
+import Icon from "metabase/components/Icon";
+import { color } from "metabase/lib/colors";
+
+export const CheckboxRoot = styled.label`
+  display: block;
+  cursor: pointer;
+
+  ${props =>
+    props.disabled &&
+    css`
+      opacity: 0.4;
+      pointer-events: none;
+    `}
+`;
+
+export const Container = styled.div`
+  display: flex;
+  align-items: center;
+`;
+
+export const VisibleBox = styled.span`
+  display: flex;
+  align-items: center;
+  justify-center: center;
+  position: relative;
+  width: ${props => `${props.size}px`};
+  height: ${props => `${props.size}px`};
+
+  background-color: ${props =>
+    props.checked ? color(props.checkedColor) : color("bg-white")};
+
+  border: 2px solid
+    ${props =>
+      props.checked ? color(props.checkedColor) : color(props.uncheckedColor)};
+
+  border-radius: 4px;
+
+  ${props =>
+    props.isFocused &&
+    !props.checked &&
+    css`
+      outline: 1px auto ${color(props.checkedColor)};
```

Comment:
When a checkbox is checked it makes the focus state not visible. Maybe it worth having the same focus outline for all colors from the UX perspective as well. What do you think?

Reply:
So maybe we won't also need to handle focus and blur manually

Reply:
@alxnddr good catch, thank you. Fixed that, so the outline is displayed for the focused state as well. Still, not sure we can easily avoid manual focus handling until we have a more fresh version of `styled-components` :(

**Demo**

![2021-07-09 13 48 33](https://user-images.githubusercontent.com/17258145/125067284-b9c37500-e0bc-11eb-8279-bce358e049f3.gif)

![2021-07-09 13 48 55](https://user-images.githubusercontent.com/17258145/125067285-baf4a200-e0bc-11eb-960b-deba4ddc0c22.gif)